src/phase0/state_transition/epoch_processing.rs

```diff
@@ -315,8 +315,18 @@ pub fn process_historical_roots_update<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    let next_epoch = get_current_epoch(state, context) + 1;
+    if (next_epoch % (SLOTS_PER_HISTORICAL_ROOT as u64 / context.slots_per_epoch)) == 0 {
```

Comment:
Specs call out the division should be a floor (`//`) -- I originally thought Rust's `u64` does this automatically, but I know Lighthouse uses some custom macros/methods called `safe_div`. And now I'm seeing some nightly/unstable methods do floor division, so I might need to refactor.

Is converting to an `f64` to use its `floor` method, and then back to a `u64` too messy?

Reply:
Okay -- so nvm, I wrote a quick test to validate (should have done this prior), and `u64` does properly round to floor. Idk if something like `safe_div` is needed, so I'll assume simply using `/` should suffice.

Reply:
right, the arithmetic operators on unsigned integer types in rust do the integer operations.

there are some corner cases to watch out for and this is why lighthouse uses the `safe_*` methods.

i am certainly open to making a "safety" pass on the code to use these sorts of methods but lean towards this repo having as little additional conceptual overhead as possible to facilitate understandability / readability.

and to that end i have been trying to stick to the plain arithmetic operators where possible.

if you feel strongly, i'd open an issue and we can move the conversation there

Reply:
All for simplicity...can do that if there's a reason one day but yeah definitely no need for it.